src/anomalib/models/image/supersimplenet/README.md

```diff
@@ -0,0 +1,57 @@
+# SuperSimpleNet: Unifying Unsupervised and Supervised Learning for Fast and Reliable Surface Defect Detection
+
+This is an implementation of the [SuperSimpleNet](https://arxiv.org/pdf/2408.03143) paper, based on the [official code](https://github.com/blaz-r/SuperSimpleNet).
+
+Model Type: Segmentation
+
+## Description
+
+**SuperSimpleNet** is a simple yet strong discriminative defect / anomaly detection model evolved from the SimpleNet architecture. It consists of four components:
+feature extractor with upscaling, feature adaptor, synthetic feature-level anomaly generation module, and
+segmentation-detection module.
+
+A ResNet-like feature extractor first extracts features, which are then upscaled and
+average-pooled to capture neighboring context. Features are further refined for anomaly detection task in the adaptor module.
+During training, synthetic anomalies are generated at the feature level by adding Gaussian noise to regions defined by the
+binary Perlin noise mask. The perturbed features are then fed into the segmentation-detection
+module, which produces the anomaly map and the anomaly score. During inference, anomaly generation is skipped, and the model
+directly predicts the anomaly map and score. The predicted anomaly map is upscaled to match the input image size
+and refined with a Gaussian filter.
+
+This implementation supports both unsupervised and supervised setting, but Anomalib currently supports only unsupervised learning.
```

Comment:
What are the missing points in Anomalib to support supervised setting

Reply:
Right now I believe there are no standard supervised datasets. Another problem is the Folder dataset as it assumes that abnormal samples are always in test set:
https://github.com/openvinotoolkit/anomalib/blob/bcc0b439f616b13a8629cb64d8bf0f88fc9083a8/src/anomalib/data/image/folder.py#L174-L175

Another thing for full reproduction of SuperSimpleNet results is the fixed flipping augmentation and frequency sampling. This is however not necessary, but needed for best results. It's also not SuperSimpleNet specific, so might be worth considering if other supervised model will be supported.

Reply:
yeah, we would like to diversify the model pool, and include more learning types than one-class models. Thanks for the feedback. 
@abc-125, you might want to be aware of this discussion as you have recently worked on this stuff

Reply:
Thank you for adding me, it would be great to have supervised models and datasets in Anomalib. Recently, I looked at how to add a supervised [dataset](https://github.com/abc-125/vad), and it certainly would require changing some base structures, such as paths to folders (I guess we will need `abnormal_train_dir` and maybe renaming the rest to make it easier to understand, `normal_train_dir`, etc.):
https://github.com/openvinotoolkit/anomalib/blob/bcc0b439f616b13a8629cb64d8bf0f88fc9083a8/src/anomalib/data/image/folder.py#L195-L202